### DISCUSSION TITLE
Styling Manager auto-hide properties

### DISCUSSION THREAD

**Discussion by josejbracho** [GrapesJS/grapesjs#5535] [Q&A]
I'm trying to show/hide certain properties on the styling manager depending on others, for example: only show flex related properties if flex is enabled in the component.

There is a default behaviour that works, whenever the properties are defined as lists I get the desired behaviour.

When flex is enabled, other properties are displayed: 
    
<img width="269" alt="Screenshot 2023-11-28 at 3 38 23 PM" src="https://github.com/GrapesJS/grapesjs/assets/83281328/60ce4d3e-6790-4b8a-a920-35a63977c67d">
    
When flex is disabled, other properties are hidden:
    
<img width="265" alt="Screenshot 2023-11-28 at 3 38 30 PM" src="https://github.com/GrapesJS/grapesjs/assets/83281328/39a8556e-12fd-4d49-89b6-e27abf5cbd2d">
    
Inspecting the elements, I can see they get a `style="display: none"` that actually hides the properties.
    
<img width="410" alt="Screenshot 2023-11-28 at 3 39 05 PM" src="https://github.com/GrapesJS/grapesjs/assets/83281328/1534948d-fa15-41ed-9564-09c3c506cc68">

If I switch the properties to be radios, they do not follow this behaviour anymore: 

<img width="268" alt="Screenshot 2023-11-28 at 3 48 05 PM" src="https://github.com/GrapesJS/grapesjs/assets/83281328/67bcdc09-ca8c-465e-be61-5a0fa277ea16">

What is driving this and how can I make sure to apply it on properties that are now radio selected?

Thanks.

**Comment by artf** [comment]
Can you share a reproducible demo?

**Reply by josejbracho** [reply]
I'm not sure I am able to replicate this on a demo, it seems like the `"style="display: none" `is added only on those classes `gjs-sm-property gjs-sm-select gjs-sm-property__align-items` as an inline property but how can I make it so the radio classes also receive this? I've gone through my code but can't find any behaviour that would allow something like this.

**Reply by josejbracho** [reply]
I have found where this is triggered. Styling properties take a `required` prop that let's you send an object to hide/show them based on the property set here.

So setting a `requires: {display: ['flex']},` on the property definition in the styleManager gets the desired behaviour.

Check this on the repo files: https://github.com/GrapesJS/grapesjs/blob/3faf9077fc584c4738d5fa3d71a80ee790dcb342/src/style_manager/model/Property.ts#L56
